HaloDAO/halodao-interface
# MiniChef V2 frontend integration

## Description of Changes
This PR combines all 3 MiniChefV2 integration tasks into one

[HDF-284 fetch pool addresses from rewards contract](https://halodao.atlassian.net/browse/HDF-284)
[HDF-285 display pool info (staked balance, pending reward)](https://halodao.atlassian.net/browse/HDF-285)
[HDF-286 stake, unstake & harvest](https://halodao.atlassian.net/browse/HDF-286)

## How to Test

Please ignore the generated firebase preview link, it is still pointing to the old `Rewards.sol` contract.

For this PR, it is encourage that you deploy the new `AmmRewards.sol` contract from the `develop` branch of `halo-rewards` repo and update your local `.env` with the new contract addresses. There are additional setup steps but if you wanna use the existing contracts I deployed, you can skip ahead to step 3 below:

1. Deploy the new contracts on `halo-rewards` > `develop` branch by running `yarn run deploy-verify:kovan`
2. Release the epoch 0 rewards and set the rewards token per second using etherscan:
    1. Approve token spend on HALO contract (spender: RewardsManager address, amount: 100K)
    2. Call releaseEpockRewards on RewardsManager contract (amount: 100K)
    3. Call setRewardTokenPerSecond on Rewards contract (amount: 100K / # of seconds in 1 month)
3. Back to this repo, change contract addresses in .env
    - Here's the contract addresses I already deployed (skips # 1 & 2):
        ```
        haloTokenContract deployed at:  0x16D185d025bF592114D1A68f83085F36159f6CdA
        halohaloContract deployed at:  0xbcbfEac78133D1efe71Ba16D4F4221b9AD4bAd01
        rewardsContract deployed at contract address  0xAa83BAc03b590e67D7217670308B392FDd55803e
        rewardsManager deployed at contract address  0x3B25a58Ca40c93F690388126478cc0F58E42Bd3C
        ```
4. Run the app using `yarn start`
5. Proceed with testing the full app features: displaying of pools, stake, unstake, harvest
6. (Optional) After seeing your rewards, you may call releaseEpockRewards on RewardsManager contract again to trigger epoch 1, this should increase the value of RNBW indicated on Vesting page

### Developer Checklist:


### Reviewers Checklist:
